tests/Tests/Query/PipelineUpdateTest.php

```diff
@@ -0,0 +1,230 @@
+        self::assertSame(2, $user2->getHits());
+    }
+
+    public function testUpdateManyWithAggregationBuilder(): void
+    {
+        $builder = $this->dm->createAggregationBuilder(User::class);
+        $builder
+            ->set()
+                ->field('hits')
+                ->multiply('$hits', 2);
+
+        $this->dm->createQueryBuilder(User::class)
+            ->updateMany()
+            ->pipeline($builder)
+            ->getQuery()
+            ->execute();
+
+        $this->dm->clear();
+
+        $user1 = $this->dm->getRepository(User::class)->find($this->user1->getId());
+        self::assertSame(2, $user1->getHits());
+
+        $user2 = $this->dm->getRepository(User::class)->find($this->user2->getId());
+        self::assertSame(4, $user2->getHits());
+    }
+
+    public function testUpdateManyWithDriverPipeline(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::set(
+                hits: Expression::multiply(Expression::fieldPath('hits'), 2),
+            ),
+        );
+
+        $this->dm->createQueryBuilder(User::class)
+            ->updateMany()
+            ->pipeline($pipeline)
+            ->getQuery()
+            ->execute();
+
+        $this->dm->clear();
+
+        $user1 = $this->dm->getRepository(User::class)->find($this->user1->getId());
+        self::assertSame(2, $user1->getHits());
+
+        $user2 = $this->dm->getRepository(User::class)->find($this->user2->getId());
+        self::assertSame(4, $user2->getHits());
+    }
+
+    public function testUpdateManyWithPipelineArray(): void
+    {
+        $this->dm->createQueryBuilder(User::class)
+            ->updateMany()
+            ->pipeline([['$set' => ['hits' => ['$multiply' => ['$hits', 2]]]]])
+            ->getQuery()
+            ->execute();
+
+        $this->dm->clear();
+
+        $user1 = $this->dm->getRepository(User::class)->find($this->user1->getId());
+        self::assertSame(2, $user1->getHits());
+
+        $user2 = $this->dm->getRepository(User::class)->find($this->user2->getId());
+        self::assertSame(4, $user2->getHits());
+    }
+
+    public function testFindOneAndUpdateWithAggregationBuilder(): void
+    {
+        $builder = $this->dm->createAggregationBuilder(User::class);
+        $builder
+            ->set()
+                ->field('hits')
+                ->add('$hits', 1);
+
+        $user = $this->dm->createQueryBuilder(User::class)
+            ->findAndUpdate()
+            ->returnNew()
+            ->field('username')->equals('foo')
+            ->pipeline($builder)
+            ->getQuery()
+            ->execute();
+
+        self::assertInstanceOf(User::class, $user);
+        self::assertSame(2, $user->getHits());
+    }
+
+    public function testFindOneAndUpdateWithDriverPipeline(): void
+    {
+        $this->markTestIncomplete('Collection::findAndUpdate does not support pipeline updates (PHPLIB-1699)');
```

Comment:
Incomplete test make the test suite fail
```suggestion
        $this->markTestSkipped('Collection::findAndUpdate does not support pipeline updates (PHPLIB-1699)');
```

Reply:
Huh, TIL that incomplete tests fail the build 🤷

Reply:
We ask for it: https://github.com/doctrine/mongodb-odm/blob/257a0a2ab0ba140b36b7be7e3f3eecb39a7c5059/phpunit.xml.dist#L13